tests/modules/seq2vec_encoders/pytorch_seq2vec_wrapper_test.py

```diff
@@ -0,0 +1,77 @@
+# pylint: disable=no-self-use,invalid-name
+import pytest
+from numpy.testing import assert_almost_equal
+import torch
+from torch.autograd import Variable
+from torch.nn import LSTM
+from torch.nn.utils.rnn import pack_padded_sequence
+
+from allennlp.modules.seq2vec_encoders import PytorchSeq2VecWrapper
+from allennlp.common.tensor import sort_batch_by_length
+from allennlp.testing.test_case import AllenNlpTestCase
+from allennlp.common.checks import ConfigurationError
+
+
+class TestPytorchSeq2VecWrapper(AllenNlpTestCase):
+    def test_get_output_dim_is_correct(self):
+        lstm = LSTM(bidirectional=True, num_layers=3, input_size=2, hidden_size=7, batch_first=True)
+        encoder = PytorchSeq2VecWrapper(lstm)
+        assert encoder.get_output_dim() == 14
+        lstm = LSTM(bidirectional=False, num_layers=3, input_size=2, hidden_size=7, batch_first=True)
+        encoder = PytorchSeq2VecWrapper(lstm)
+        assert encoder.get_output_dim() == 7
+
+    def test_forward_pulls_out_correct_tensor_without_sequence_lengths(self):
+        lstm = LSTM(bidirectional=True, num_layers=3, input_size=2, hidden_size=7, batch_first=True)
+        encoder = PytorchSeq2VecWrapper(lstm)
+        input_tensor = Variable(torch.FloatTensor([[[.7, .8], [.1, 1.5]]]))
+        lstm_output = lstm(input_tensor)
+        encoder_output = encoder(input_tensor)
+        assert_almost_equal(encoder_output.data.numpy(), lstm_output[0].data.numpy()[:, -1, :])
+
+    def test_forward_pulls_out_correct_tensor_with_sequence_lengths(self):
+        lstm = LSTM(bidirectional=True, num_layers=3, input_size=3, hidden_size=7, batch_first=True)
+        encoder = PytorchSeq2VecWrapper(lstm)
+
+        tensor = torch.rand([5, 7, 3])
+        tensor[1, 6:, :] = 0
+        tensor[2, 4:, :] = 0
+        tensor[3, 2:, :] = 0
+        tensor[4, 1:, :] = 0
+
+        input_tensor = Variable(tensor)
+        sequence_lengths = torch.LongTensor([7, 6, 4, 2, 1])
+        packed_sequence = pack_padded_sequence(input_tensor, list(sequence_lengths), batch_first=True)
+        _, state = lstm(packed_sequence)
+        # Transpose output state, extract the last forward and backward states and
+        # reshape to be of dimension (batch_size, 2 * hidden_size).
+        reshaped_state = state[0].transpose(0, 1)[:, -2:, :].contiguous().view(-1, 14)
```

Comment:
Can we test against an explicit concatenation, instead of a reshaping?  It'd be nice to be absolutely sure that the reshaping is happening as we expect it is.